core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -919,7 +924,11 @@ public Result visit(TableFunctionScan e) {
     final String lowerName = name.toLowerCase(Locale.ROOT);
     if (lowerName.startsWith("expr$")) {
       // Put it in ordinalMap
-      ordinalMap.put(lowerName, node);
+      if (stack.size() != 1) {
+        node = as(node, name);
```

Comment:
I understand this as that we don't need to add an alias for the outmost node.

Reply:
The semantic is correct, but, can we have a more simple solution for the SQL, the `RelToSqlConverterTest` has many failing test cases(some of which is correct to fix but most are unnecessary change)

Reply:
Agree with Danny, +1